src/django_components/component.py

```diff
@@ -126,15 +203,17 @@ def _resolve_component_relative_files(attrs: MutableMapping) -> None:
     # Check if filepath refers to a file that's in the same directory as the component class.
     # If yes, modify the path to refer to the relative file.
     # If not, don't modify anything.
-    def resolve_file(filepath: str) -> str:
-        maybe_resolved_filepath = os.path.join(comp_dir_abs, filepath)
-        component_import_filepath = os.path.join(comp_dir_rel, filepath)
-
-        if os.path.isfile(maybe_resolved_filepath):
-            logger.debug(
-                f"Interpreting template '{filepath}' of component '{module_name}' relatively to component file"
-            )
-            return component_import_filepath
+    def resolve_file(filepath: str | SafeData) -> str | SafeData:
+        if isinstance(filepath, str):
```

Comment:
Path resolution logic is applied only to paths that resolved to plain string (so paths defined as str, bytes, PathLike, functions).